org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateDomainSpecificationConstants.mtl

```diff
@@ -41,14 +41,14 @@ endif)
 
 [query public resourceTypeLocalName(aResource: Resource) : String =
 (if (aResource.describes.oclIsUndefined()) then
-    '"'.concat(aResource.javaName(true)).concat('"')
+    '"'.concat(aResource.name).concat('"')
 else
-	'"'.concat(aResource.describes.javaName(true)).concat('"')
+	'"'.concat(aResource.describes.name).concat('"')
 endif)
 /]
 
 [template public generateResourceConstants(aResource : Resource)]
-public static String [resourcePathConstantName(aResource)/] = "[aResource.javaName(false)/]";
+public static String [resourcePathConstantName(aResource)/] = "[aResource.name.toLowerFirst()/]";
```

Comment:
why convert to lowercase? e.g. http://dublincore.org/documents/dcmi-terms/#terms-isVersionOf shall preserve the case

Reply:
This is the url path for GET on the resource shape. I thought it neater is the url is lower case. 

For the rdf localName, See 2 lines down.

Reply:
Have a look at existing generated code. The plan was not to change behavior here.